src/containers/pages/AssigmentMapWrapper/helpers/utils.tsx

```diff
@@ -0,0 +1,110 @@
+import { Feature } from '@turf/turf';
+import { History } from 'history';
+import { EventData, Map } from 'mapbox-gl';
+import { AssignmentMapWrapperProps } from '..';
+import { PLAN_TYPES_WITH_MULTI_JURISDICTIONS } from '../../../../configs/env';
+import {
+  AUTO_ASSIGN_JURISDICTIONS_URL,
+  MANUAL_ASSIGN_JURISDICTIONS_URL,
+} from '../../../../constants';
+import { SELECTION_REASON } from '../../../../store/ducks/opensrp/hierarchies/constants';
+import { TreeNode } from '../../../../store/ducks/opensrp/hierarchies/types';
+import { nodeIsSelected } from '../../../../store/ducks/opensrp/hierarchies/utils';
+import { UseContext } from '../../../../store/ducks/plans';
+
+/**
+ * Handler to get current parent id from clicked jurisdiction on map
+ * @param props - AssignmentMapWrapper component props
+ * @param setMapParent - function to set map parent id to state
+ * @param history - history session object
+ */
+
+export const onJurisdictionClick = (
+  props: AssignmentMapWrapperProps,
+  setMapParent: (currentId: string) => void,
+  history: History
+) => {
+  function handleJurisdictionClick(_: Map, e: EventData) {
+    // Destructure event data
+    const { point, target, originalEvent } = e;
+    const {
+      currentParentNode,
+      rootJurisdictionId,
+      selectNodeCreator,
+      deselectNodeCreator,
+      plan,
+      autoSelectionFlow,
+      getJurisdictionsMetadata,
+    } = props;
+
+    const { useContext } = plan;
+    const getInterventionType = useContext.find(
+      context => context.code === 'interventionType'
+    ) as UseContext;
```

Comment:
lets use [this](https://github.com/onaio/reveal-frontend/blob/bd679d507b221c883245ad095c144c263811583a/src/helpers/utils.tsx#L882)